packages/vm/src/runBlock.ts

```diff
@@ -53,8 +45,8 @@ export async function runBlock(this: VM, opts: RunBlockOpts): Promise<RunBlockRe
 
   if (
     this._hardforkByBlockNumber ||
-    isTruthy(this._hardforkByTTD) ||
-    isTruthy(opts.hardforkByTTD)
+    this._hardforkByTTD !== undefined ||
```

Comment:
In `vm.ts` this field is marked as optional, but it looks to me that it is always being set? I think we can improve here by making that field non-optional?

Reply:
I tried making this.hardforkByTTD non-optional (which ends up removing that if altogether) but ended up with a couple failing API tests. I've created an issue for it in case we want to investigate that further (#2260)

Reply:
No no, just to add here: there are a lot of development use cases where this is not wanted. So these hardforkBy* options should always be present and available to set.

Reply:
The type of `hardforkByTTD` is `undefined | bigint`, and with `optional` I meant that it should be of type `bigint` (so it should never be `undefined`), but this ruins the logic of setting the hardforks (I realized this when checking #2260). I'm not sure what you mean "should always be available to set", if you want to set the `hardforkByBlockNumber` from `false` to `true` or vice-versa you have to either hack it in, or you have to create a new VM since it is `readonly`.

Reply:
Plainly meant this should remain an option (in the case this would be in question). 🙂